File: app/src/main/java/com/neome/api/meta/base/dto/StudioEntSpreadsheet.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import org.jetbrains.annotations.Nullable;

import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdForm;
import com.neome.api.meta.base.Types.MetaIdRole;
import com.neome.api.meta.base.Types.MetaIdSpreadsheet;
import com.neome.api.meta.base.Types.MetaIdVar;
import com.neome.api.meta.base.dto.StudioBase;
import com.neome.api.meta.base.dto.StudioDetails;
import com.neome.api.meta.base.dto.StudioMapOfForwardRolePermission;
import com.neome.api.meta.base.dto.StudioMapOfLayoutGrid;
import com.neome.api.meta.base.dto.StudioMapOfPartition;

@SuppressWarnings("unused")
public class StudioEntSpreadsheet extends StudioBase
{
  public String alias;

  @Nullable
  public MetaIdRole[] auditTrailFormRoleSet;

  @Nullable
  public MetaIdRole[] bypassDurationRoleSet;

  @Nullable
  public MetaIdRole[] clearRoleSet;

  public StudioDetails details;

  public MetaIdForm formId;

  @Nullable
  public StudioMapOfForwardRolePermission groupForwardRolePermissionMap;

  @Nullable
  public MetaIdField[] historyIdSet;

  @Nullable
  public MetaIdRole[] insertRoleSet;

  @Nullable
  public MetaIdVar invisibleAfterDurationVarId;

  @Nullable
  public Boolean isMasterSheet;

  @Nullable
  public StudioMapOfLayoutGrid layoutSpreadsheetMap;

  public MetaIdSpreadsheet metaId;

  @Nullable
  public StudioMapOfPartition partitionMap;

  @Nullable
  public MetaIdField[] queryableIdSet;

  @Nullable
  public MetaIdField[] ragIdSet;

  @Nullable
  public MetaIdVar readAfterDurationVarId;

  @Nullable
  public MetaIdRole[] readRoleSet;

  @Nullable
  public MetaIdVar readRoleVarId;

  @Nullable
  public MetaIdRole[] removeRoleSet;

  @Nullable
  public MetaIdField[] searchableIdSet;

  @Nullable
  public Boolean supportOffline;

  @Nullable
  public MetaIdField[] uniqueIdSet;

  @Nullable
  public MetaIdField[] uniqueIndexIdSet;

  @Nullable
  public MetaIdRole[] updateRoleSet;

  @Nullable
  public StudioMapOfForwardRolePermission userForwardRolePermissionMap;
}

File: app/src/main/java/com/neome/api/meta/base/dto/StudioEntSpreadsheet.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnUniquenessMode
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdForm
import com.neome.api.meta.base.Types.MetaIdRole
import com.neome.api.meta.base.Types.MetaIdSpreadsheet
import com.neome.api.meta.base.Types.MetaIdVar
import com.neome.api.meta.base.dto.StudioBase
import com.neome.api.meta.base.dto.StudioDetails
import com.neome.api.meta.base.dto.StudioMapOfForwardRolePermission
import com.neome.api.meta.base.dto.StudioMapOfLayoutGrid
import com.neome.api.meta.base.dto.StudioMapOfPartition

interface StudioEntSpreadsheet : StudioBase
{
  val alias: String
  val auditTrailFormRoleSet: List<MetaIdRole>?
  val bypassDurationRoleSet: List<MetaIdRole>?
  val clearRoleSet: List<MetaIdRole>?
  val details: StudioDetails
  val formId: MetaIdForm
  val groupForwardRolePermissionMap: StudioMapOfForwardRolePermission?
  val historyIdSet: List<MetaIdField>?
  val insertRoleSet: List<MetaIdRole>?
  val invisibleAfterDurationVarId: MetaIdVar?
  val isMasterSheet: Boolean?
  val layoutSpreadsheetMap: StudioMapOfLayoutGrid?
  val metaId: MetaIdSpreadsheet
  val partitionMap: StudioMapOfPartition?
  val queryableIdSet: List<MetaIdField>?
  val ragGroupId: MetaIdField?
  val ragIdSet: List<MetaIdField>?
  val readAfterDurationVarId: MetaIdVar?
  val readRoleSet: List<MetaIdRole>?
  val readRoleVarId: MetaIdVar?
  val removeRoleSet: List<MetaIdRole>?
  val searchableIdSet: List<MetaIdField>?
  val supportOffline: Boolean?
  val uniqueIdSet: List<MetaIdField>?
  val uniqueIndexIdSet: List<MetaIdField>?
  val uniquenessMode: EnumDefnUniquenessMode?
  val updateRoleSet: List<MetaIdRole>?
  val userForwardRolePermissionMap: StudioMapOfForwardRolePermission?
}